### PR TITLE
 Add case for Hot-plug nic w/ the netdev already used

### PR DESCRIPTION
**Pull request by weliao**
Signed-off-by: Wei Liao weliao@redhat.com

ID: 1390116

### CODE REVIEW

**Comment by weliao** [comment]
TESTS      : 3
 (1/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.nic_hotplug.used_netdev.nic_8139: PASS (88.30 s)
 (2/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.nic_hotplug.used_netdev.nic_virtio: PASS (45.97 s)
 (3/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.nic_hotplug.used_netdev.nic_e1000: PASS (50.45 s)

File: qemu/tests/nic_hotplug.py [hunk 1]
```diff
@@ -115,84 +124,95 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
...
+            used_sameid = params.get("used_sameid")
+            if used_sameid == "yes":
+                useddevice_id = primary_nic[0].netdev_id
+                error.context("Hot-plug nic w/ the netdev already used", logging.info)
+                try:
+                    add_output = device_add_nic(nic_model, useddevice_id, nic_name)
+                except Exception, e:
+                    e = str(e)
+                    match_error = "it's in use"
+                    if match_error not in e:
```
Comment (xutian) [inline qemu/tests/nic_hotplug.py]: why not
`if match_err not in str(e):`

Comment (xutian) [inline qemu/tests/nic_hotplug.py]: And I think logical should be

```
try:
     do something
except Exception, e:
    if match_error in str(e):
        pass
    else:
        raise 
```

File: qemu/tests/nic_hotplug.py [hunk 2]
```diff
@@ -115,84 +124,95 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
...
+
+                error.context("Ping guest's new ip from host", logging.info)
+                status, output = ping_hotplug_nic(host_ip_addr, hotplug_nic["mac"],
+                                                  s_session, guest_is_linux)
+                if status:
+                    err_msg = "New nic failed ping test, error info: '%s'"
+                    raise error.TestFail(err_msg % output)
+
+                error.context("Pause vm", logging.info)
+                vm.monitor.cmd("stop")
```
Comment (xutian) [inline qemu/tests/nic_hotplug.py]: why not call vm.pause and vm.resume.

Comment (xutian) [inline qemu/tests/nic_hotplug.py]: I think better  to pause VM  a long time(eg, 600s) then resume it and not pause it, then resume it immediately

File: qemu/tests/nic_hotplug.py [hunk 3]
```diff
@@ -115,84 +124,95 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
...
+                vm.verify_status("paused")
+                error.context("Resume vm", logging.info)
+                vm.monitor.cmd("cont")
+                vm.verify_status("running")
+                error.context("Ping guest's new ip after resume", logging.info)
```
Comment (xutian) [inline qemu/tests/nic_hotplug.py]: do you need refresh guest route table after  unplug operation?

File: qemu/tests/nic_hotplug.py (hunk 2 above)
Comment (weliao) [inline qemu/tests/nic_hotplug.py]: this is old code, I didn't write it

File: qemu/tests/nic_hotplug.py (hunk 3 above)
Comment (weliao) [inline qemu/tests/nic_hotplug.py]: this is old code, I didn't write it

**Comment by CongLi** [comment]
@vivianQizhu will help update this issue later.

**Pull request closed without merging**